numba/_lapack.c

```diff
@@ -486,12 +507,52 @@ numba_xxpotrf(char kind, char uplo, Py_ssize_t n, void *a, Py_ssize_t lda)
     _lda = (F_INT) lda;
 
     (*(xxpotrf_t) raw_func)(&uplo, &_n, a, &_lda, &info);
-    CATCH_LAPACK_INVALID_ARG(info);
+    CATCH_LAPACK_INVALID_ARG("xxpotrf", info);
     return info;
 }
 
 
 /*
+ * kind_size()
+ * gets the data size appropriate for a specified kind.
+ *
+ * Input:
+ * kind - the kind, one of:
+ *         (s, d, c, z) = (float, double, complex, double complex).
+ *
+ * data_size - modified in place, on return contains the appropriate data size.
+ *
+ * Returns 0 on success, -1 else.
```

Comment:
Right, well perhaps in fact all these switches over `kind` (which occur in a lot of places) ought just not have a default and instead we have a guard function (could be macro'd) as the first line in all these functions, check is done once then?